app/src/pages/settings/NewUserDialog.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback } from "react";
+import { graphql, useMutation } from "react-relay";
+
+import { Dialog, DialogContainer } from "@arizeai/components";
+
+import {
+  UserForm,
+  UserFormParams,
+} from "@phoenix/components/settings/UserForm";
+
+import { NewUserDialogMutation } from "./__generated__/NewUserDialogMutation.graphql";
+
+export function NewUserDialog({
+  onNewUserCreated,
+  onNewUserCreationError,
+  onDismiss,
+}: {
+  onNewUserCreated: (email: string) => void;
+  onNewUserCreationError: (error: Error) => void;
+  onDismiss: () => void;
+}) {
+  const [commit, isCommitting] = useMutation<NewUserDialogMutation>(graphql`
+    mutation NewUserDialogMutation($input: CreateUserInput!) {
+      createUser(input: $input) {
+        user {
+          id
+          email
+        }
```

Comment:
@mikeldking i did not have luck with datasets in spreading fragments that fetched connections and getting the fragment to refetch so i did not add query type onto the return here and am instead using fetchKey - let me know if i'm missing someting but connections don't seem to be refetchable like that? https://relay.dev/docs/tutorial/mutations-updates/#adding-comments--mutations-on-connections - although users table doesn' thave the connection directive so maybe it would work here...

Reply:
Yeah I'm struggling with this as well. We can maybe use relay's optimistic updates instead. Then things will be better encapsulated

Reply:
yeah i'm not sure if optimistic updates work for connections, i might just try the connection directives - i had trouble last time but maybe i can get it to work here